kanidm_book/src/examples/k8s_ingress_example.md

```diff
@@ -0,0 +1,279 @@
+kanidm system oauth2 create_scope_map "$name" k8s_example_group openid email profile --name admin
+kanidm system oauth2 get "$name" --name admin
+
+## For <cookie_secret>
+echo "Generating cookie secret:"
+docker run -ti --rm python:3-alpine python -c 'import secrets,base64; print(base64.b64encode(base64.b64encode(secrets.token_bytes(16))).decode("utf-8"));'
+```
+
+2. Create a file called `k8s.kanidm-nginx-auth-example.yaml` with the block below, replacing every `<string>` (drop the `<>`) with appropriate values.
+
+```yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kanidm-example
+  labels:
+    pod-security.kubernetes.io/enforce: restricted
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: kanidm-example
+  name: my-homepage
+  labels:
+    app: my-homepage
+spec:
+  revisionHistoryLimit: 1
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-homepage
+  template:
+    metadata:
+      labels:
+        app: my-homepage
+    spec:
+      containers:
+        - name: my-hompage
+          image: modem7/docker-starwars
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop: ["ALL"]
+      securityContext:
+        runAsNonRoot: true
+        seccompProfile:
+          type: RuntimeDefault
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: kanidm-example
+  name: my-homepage
+spec:
+  selector:
+    app: my-homepage
+  ports:
+    - protocol: TCP
+      port: 8080
+      targetPort: 8080
+
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    cert-manager.io/cluster-issuer: lets-encrypt-cluster-issuer
+    nginx.ingress.kubernetes.io/auth-url: "https://$host/oauth2/auth"
+    nginx.ingress.kubernetes.io/auth-signin: "https://$host/oauth2/start?rd=$escaped_request_uri"
+  name: my-homepage
+  namespace: kanidm-example
+spec:
+  ingressClassName: nginx
+  tls:
+    - hosts:
+        - <hostname>
+      secretName: <hostname>-ingress-tls # replace . with - in the hostname
+  rules:
+  - host: <hostname>
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: my-homepage
+            port:
+              number: 8080
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    k8s-app: oauth2-proxy-kanidm-example
+  name: oauth2-proxy-kanidm-example
+  namespace: kanidm-example
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: oauth2-proxy-kanidm-example
+  template:
+    metadata:
+      labels:
+        k8s-app: oauth2-proxy-kanidm-example
+    spec:
+      containers:
+      - args:
+        - --provider=oidc
+        - --email-domain=*
+        - --upstream=file:///dev/null
+        - --http-address=0.0.0.0:4182
+        - --oidc-issuer-url=https://<kanidm-domain>/oauth2/openid/<oauth2_rs_name>
```

Comment:
We could make these `<kanidm-domain>` in upper case to make them more obvious, like `<KANIDM-DOMAIN>`